modules/uhttp/client.go

```diff
@@ -0,0 +1,100 @@
+package uhttp
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/net/context/ctxhttp"
+
+	"go.uber.org/fx/core"
+)
+
+// Client wraps around a http client
+type Client struct {
+	http.Client
+	filters []ClientFilter
+}
+
+// NewClient creates a new instance of uhttp Client
+func NewClient(client http.Client, filters ...ClientFilter) *Client {
+	if filters == nil {
+		filters = []ClientFilter{ClientFilterFunc(tracingClientFilter)}
+	}
+	return &Client{Client: client, filters: filters}
```

Comment:
Would we ever not allow a tracing filter? Shouldn't we just unilaterally append it?